tests/unit/api_test.py

```diff
@@ -544,17 +544,17 @@ def frame_header(stream, data):
 
     def request(self, stream=None, tty=None, demux=None):
         assert stream is not None and tty is not None and demux is not None
-        with APIClient(
+        client = APIClient(
                 base_url=self.address,
                 version=DEFAULT_DOCKER_API_VERSION
-                ) as client:
-            if tty:
-                url = client._url('/tty')
-            else:
-                url = client._url('/no-tty')
-            resp = client._post(url, stream=True)
-            return client._read_from_socket(
-                resp, stream=stream, tty=tty, demux=demux)
+                )
+        if tty:
+            url = client._url('/tty')
+        else:
+            url = client._url('/no-tty')
+        resp = client._post(url, stream=True)
+        return client._read_from_socket(
+            resp, stream=stream, tty=tty, demux=demux)
```

Comment:
Is this deliberately introducing a leak by lifting the APIClient out of a `with`?

Would it be better to make this a generator instead or something?